### DISCUSSION TITLE
Slice volume change the name of the axes

### DISCUSSION THREAD

**Discussion by DVangelis** [marcomusy/vedo#674] [Q&A] [answered]
Hi I am new to vedo. 
I try to plot 3d stacks with using the module Slicer3DPlotter. I would like to rename the axis where it is Z I would like to name it X and so on. How I could do that maybe trough calling the vedo.show() ? Also is there a way to record in a video of the 3d slices moving up and down. I tried the vedo.show().record() but jupyter notebook was crashing. 
Thanks in advance

**Comment by marcomusy** [comment]
Hi, the `vedo.show().record() ` cannot work because the `Slicer3DPlotter` is already a `Plotter` object.

Try out this:
```python
from vedo import dataurl, Volume, Text2D, Axes, Video
from vedo.applications import Slicer3DPlotter

filename = dataurl+'embryo.slc'
vol = Volume(filename)

plt = Slicer3DPlotter(
    vol,
    bg='white', bg2='lightblue',
    cmaps=("gist_ncar_r","jet","Spectral_r","hot_r","bone_r"),
    useSlider3D=False,
    axes=0,
    interactive=False,
)

axes = Axes(
    xrange=vol.xbounds(),
    yrange=vol.ybounds(),
    zrange=vol.zbounds(),
    xtitle="Y-axis",
    ytitle="Z-axis",
    ztitle="X-axis",
    xyGrid=False,
)

#Can now add any other object to the Plotter scene:
plt += axes
plt += Text2D(f'File: {filename}')

vid = Video("slicing_volume.mp4", duration=5)  # on windows this might not work

# extract the sliders
sld1, sld2, sld3 = plt.sliders
for i in range(200):
    widget, func = sld1
    widget.GetRepresentation().SetValue(i)
    func(widget, "")
    plt.camera.Azimuth(0.1) # optional
    plt.render()
    vid.addFrame()
vid.close()

plt.interactive().close()
```

https://user-images.githubusercontent.com/32848391/181051622-eda9806f-3802-4734-9bb5-67808a4ea081.mp4

**Reply by DVangelis** [reply]
Ok thanks. I just test the code. It somehow works in win10 jupyter notebook sometimes the kenrel crahses . If I want to change the slice that is moving  I just change what is in the line  below correct ?
    widget, func = sld1

. Do you recommend any other python ide for running in it that maybe it does not crash  so often?
Maybe a silly question but how I can save the video in adirectory  of my choice ,should I use the vedo.io.write it seems that mp4 format its not supported ? Thank you very much for the code and your help it really helped.

**Comment by marcomusy** [comment, marked as answer]
> If I want to change the slice that is moving I just change what is in the line below correct ?

yes

About the crashing, you may try with the latest dev-version 
`pip install -U git+https://github.com/marcomusy/vedo.git`

and/or with the spyder IDE?

About saving the file... if it doesn't work on windows I have no other solution than screen-grabbing or creating a series of screenshots in the loop with
`plt.screenshot(f"image_{i}.png")`
and then merging them with some external software.

**Reply by DVangelis** [reply]
With Spyder worked magically !!! I was able to save the video . Thank you very much.

**Reply by DVangelis** [reply]
I would like to ask one more thing. I chose to save the screenshots as the video was crashing maybe because of my computer specs ? not enough memory or the rendering was slow ? Anyways now that I save the printscreens how can I change the location of my volume inside the scene. It seems that one part of the volume the left one is cropped and does not shown of , you can see it only for later timesteps. Also I comment the line of camera azimuth in order to avoid the cropping but it did not help.

**Comment by marcomusy** [comment]
Uhm sorry I'm not sure I understand what the problem is.. what if you add instead:
`    plt.resetCamera().render()`

**Reply by DVangelis** [reply]
Ok and last thing that I  could not figure it out. Is there a way so the name of the cmap is not shown in the bottom of the render ?

**Reply by marcomusy** [reply]
You can add
```
plt.buttons[0].actor.SetInput("")
```
